.github/workflows/merge-to-stage.js

```diff
@@ -0,0 +1,233 @@
+**CC:** https://www.stage.adobe.com/creativecloud.html
+**Blog:** https://blog.stage.adobe.com/
+**Acrobat:** https://www.stage.adobe.com/acrobat/online/sign-pdf.html
+
+**Milo:**
+- Before: https://main--milo--adobecom.hlx.live/?martech=off
+- After: https://stage--milo--adobecom.hlx.live/?martech=off
+`;
+
+const RCPDates = [
+  {
+    start: new Date('2024-05-26T00:00:00-07:00'),
+    end: new Date('2024-06-01T00:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-06-13T11:00:00-07:00'),
+    end: new Date('2024-06-13T14:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-06-30T00:00:00-07:00'),
+    end: new Date('2024-07-06T00:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-08-25T00:00:00-07:00'),
+    end: new Date('2024-08-31T00:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-09-12T11:00:00-07:00'),
+    end: new Date('2024-09-12T14:00:00-07:00'),
+  },
+  {
+    start: new Date('2024-10-14T00:00:00-07:00'),
+    end: new Date('2024-11-18T17:00:00-08:00'),
+  },
+  {
+    start: new Date('2024-11-17T00:00:00-08:00'),
+    end: new Date('2024-11-30T00:00:00-08:00'),
+  },
+  {
+    start: new Date('2024-12-12T11:00:00-08:00'),
+    end: new Date('2024-12-12T14:00:00-08:00'),
+  },
+  {
+    start: new Date('2024-12-15T00:00:00-08:00'),
+    end: new Date('2025-01-02T00:00:00-08:00'),
+  },
+];
+
+const isHighPrio = (label) => label.includes(labels.highPriority);
+
+const hasFailingChecks = (checks) =>
+  checks.some(
+    ({ conclusion, name }) =>
+      name !== 'merge-to-stage' && conclusion === 'failure'
+  );
+
+const getPRs = async () => {
+  let prs = await github.rest.pulls
+    .list({ owner, repo, state: 'open', per_page: 100, base: stage })
+    .then(({ data }) => data);
+  await Promise.all(prs.map((pr) => addLabels({ pr, github, owner, repo })));
+  prs = prs.filter((pr) => pr.labels.includes(labels.readyForStage));
+  await Promise.all([
+    ...prs.map((pr) => addFiles({ pr, github, owner, repo })),
+    ...prs.map((pr) => getChecks({ pr, github, owner, repo })),
+    ...prs.map((pr) => getReviews({ pr, github, owner, repo })),
+  ]);
+
+  prs = prs.filter(({ checks, reviews, html_url, number, title }) => {
+    if (hasFailingChecks(checks)) {
+      slackNotification(slack.failingChecks({ html_url, number, title }));
+      if (number === currPrNumber)
+        core.setFailed(`Failing checks on the current PR ${number}`);
+      return false;
+    }
+
+    const approvals = reviews.filter(({ state }) => state === 'APPROVED');
+    if (approvals.length < requiredApprovals) {
+      slackNotification(slack.requireApprovals({ html_url, number, title }));
+      if (number === currPrNumber)
+        core.setFailed(`Insufficient approvals on the current PR ${number}`);
+      return false;
+    }
+
+    return true;
+  });
+
+  return prs.reverse(); // OLD PRs first
+};
+
+const merge = async ({ prs }) => {
+  console.log(`Merging ${prs.length || 0} PRs that are ready... `);
```

Comment:
Yup, also needed! It's very confusing to understand what happens, if the action just runs without any clear feedback

Reply:
A bit of context while I'm at it: 
It's written in a way to be debug-able locally with a personal access token. The token is a personal access token is restricted to opening PRs, it can't merge anything.
So debugging is a mix of local + pushing the action to your fork and testing it on Github